pkg/controller/directvolumemigration/rsync.go

```diff
@@ -145,11 +145,12 @@ func (t *Task) createRsyncConfig() error {
 		return err
 	}
 
-	err = t.createRsyncPassword(destClient)
+	err = t.createRsyncPassword()
 	if err != nil {
 		return err
 	}
-	password, err := t.getRsyncPassword(destClient)
+	time.Sleep(time.Second)
```

Comment:
Not sure on waiting for a Second or a few hundred milliseconds.

Reply:
@JaydipGabani I think we should not have a sleep in the middle of a reconcile loop. Lets get on a call to talk about this

Reply:
I would recommend you change the implementation here as follows:

1. get rsync secret, if you find it, grab the password and use it
2. if the secret is not found, create the password, create the secret and exit the reconcile loop with immediate requeue.

That will remove the sleep in the middle of the reconcile loop.